account_cash_discount_tax/wizards/account_invoice_cash_tax_wizard.py

```diff
@@ -0,0 +1,144 @@
+        :return:
+        """
+        for invoice_tax in self.invoice_id.tax_line_ids:
+            # Create one line without tax
+            amount = invoice_tax.base *\
+                (self.invoice_id.discount_percent / 100)
+            line_add = {
+                'name': self.description,
+                'price_unit': amount,
+                'account_id': self.account_id.id,
+                'quantity': 1.0,
+                'is_cash_discount_tax': True,
+            }
+            line_rem = {
+                'name': self.description,
+                'price_unit': amount,
+                'account_id': self.account_id.id,
+                'quantity': -1.0,
+                'is_cash_discount_tax': True,
+                'invoice_line_tax_ids': [(6, 0, [invoice_tax.tax_id.id])],
+            }
+            lines.append((0, 0, line_add), )
+            lines.append((0, 0, line_rem))
+
+    @api.multi
+    def _get_invoice_lines_taxes(self, lines):
+        """
+        Parse invoice line taxes
+        :param lines:
+        :return:
+        """
+        taxes = self.invoice_id.invoice_line_ids.mapped('invoice_line_tax_ids')
+        amount_per_base_tax = {}
+        for line in self.invoice_id.invoice_line_ids:
+            for line_tax in line.invoice_line_tax_ids:
+                current = amount_per_base_tax.get(line_tax.id)
+                amount = (current + line.price_subtotal) if current else\
+                    line.price_subtotal
+                amount_per_base_tax.update({
+                    line_tax.id: amount
+                })
+        for tax in taxes:
+            amount = amount_per_base_tax.get(tax.id) *\
+                (self.invoice_id.discount_percent / 100)
+            line_add = {
+                'name': self.description,
+                'price_unit': amount,
+                'account_id': self.account_id.id,
+                'quantity': 1.0,
+                'is_cash_discount_tax': True,
+            }
+            line_rem = {
+                'name': self.description,
+                'price_unit': amount,
+                'account_id': self.account_id.id,
+                'quantity': -1.0,
+                'is_cash_discount_tax': True,
+                'invoice_line_tax_ids': [(6, 0, [tax.id])],
+            }
+            lines.append((0, 0, line_add), )
+            lines.append((0, 0, line_rem))
+
+    @api.multi
+    def _prepare_invoice_lines(self):
+        """
+        We parse all taxes through line taxes (invoice_id.tax_line_ids could
+        have been generated through a tax group - but we don't want individual
+        ones)
+        :return: list
+        """
+        self.ensure_one()
+        invoice_taxes = self.invoice_id.company_id.cash_discount_invoice_tax
+        lines = []
+        if invoice_taxes:
+            self._get_invoice_taxes(lines)
+        else:
+            self._get_invoice_lines_taxes(lines)
+        return lines
+
+    @api.multi
+    def doit(self):
```

Comment:
Could you change this method name to a more indicative name like `apply` or something?